tests/test_lua_consumer.py

```diff
@@ -0,0 +1,149 @@
+import pytest
+
+from comdb2 import dbapi2
+from comdb2 import factories
+
+MONITOR_PROCEDURE = """
+local function main()
+    db:column_type('int', 1)
+    db:column_name('id', 1)
+
+    db:column_type('text', 2)
+    db:column_name('old_message', 2)
+
+    db:column_type('text', 3)
+    db:column_name('new_message', 3)
+
+    local consumer = db:consumer()
+    while true do
+        local change = consumer:get()
+        local row = {id=nil, old_message=nil, new_message=nil}
+        if change.new ~= nil then
+            row.id = change.new.id
+            row.new_message = change.new.message
+        end
+        if change.old ~= nil then
+            row.id = change.old.id
+            row.old_message = change.old.message
+        end
+
+        consumer:emit(row)
+        consumer:consume()
+    end
+end
+"""
+
+
+@pytest.fixture(autouse=True)
+def create_consumer():
+    conn = dbapi2.Connection("mattdb", "dev", autocommit=True)
+    cursor = conn.cursor()
+
+    # Drop existing consumer/procedure/table (just in case)
+    try:
```

Comment:
if this is Python 3 you can use [suppress ](https://docs.python.org/3/library/contextlib.html#contextlib.suppress) which IMHO reads nicer as it states the purpose of wanting to ignore the exception more clearly.

Reply:
That is nicer, and I thought about it, but we still have an (unsupported) 2.7 build internally, and that hasn't been spun off into a maintenance branch yet.

On the other hand, I've been meaning to split that off, so I could do that first and then use `contextlib.suppress` here. But on the other other hand, this is slightly ugly, but it's just in tests.

Reply:
That's why I started with "if this is Python 3" :)